### DISCUSSION TITLE
Plugin system for stitches

### DISCUSSION THREAD

**Discussion by behnammodi** [stitchesjs/stitches#585] [Ideas]
Hi man, @jonathantneal
This issued opened base on Discord discussion.

We want to discuss the plugin system in stitches and investigate all pros and cons

We need something like this:
```js
import { setup } from '@stitches/react';

const logPlugin = {
  onProcess: (style) => {
    console.log(style);
    
    /* 
      we can mutate everything into style and return it
    */
    return style;
  }
};

setup({ plugins: [logPlugin] });

```

**Comment by peduarte** [comment]
@behnammodi Thanks for raising this! I'm moving this to a discussion because it's not an actionable issue yet. Will need a lot fo thinking, spec'ing, define use cases, etc.
